tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+			Covers: []string{
+				".Values.egressGateways.enabled",
+				".Values.egressGateways.podDisruptionBudget",
+			},
+
+			Checks: test.Checks{
+				checker.PerResource(func(tc *checker.TestContext, pdb *policyv1beta1.PodDisruptionBudget) {
+					egressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.egressGateways.enabled")
+
+					if !egressGatewaysEnabled {
+						return
+					}
+
+					egressGatewaysPodDisruptionBudget, _ := checker.RenderValue[string](tc, ".Values.egressGateways.podDisruptionBudget")
+					assert.Equal(tc.T, egressGatewaysPodDisruptionBudget, pdb.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.ingressGateways.type",
+			Covers: []string{
+				".Values.ingressGateways.enabled",
+				".Values.ingressGateways.type",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					ingressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.ingressGateways.enabled")
+
+					if !ingressGatewaysEnabled {
+						return
+					}
+
+					ingressGatewaysType, _ := checker.RenderValue[string](tc, ".Values.ingressGateways.type")
+
+					if ingressGatewaysType != "NodePort" {
+						return
+					}
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.ingressGateways.hpaSpec",
+			Covers: []string{
+				".Values.ingressGateways.enabled",
+				".Values.ingressGateways.hpaSpec",
+			},
+
+			Checks: test.Checks{
+				checker.PerResource(func(tc *checker.TestContext, hpa *autoscalingv2.HorizontalPodAutoscaler) {
+					ingressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.ingressGateways.enabled")
+
+					if !ingressGatewaysEnabled {
+						return
+					}
+
+					ingressGatewaysHpaSpec, _ := checker.RenderValue[string](tc, ".Values.ingressGateways.hpaSpec")
+					assert.Equal(tc.T, ingressGatewaysHpaSpec, hpa.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.ingressGateways.podDisruptionBudget",
+			Covers: []string{
+				".Values.ingressGateways.enabled",
+				".Values.ingressGateways.podDisruptionBudget",
+			},
+
+			Checks: test.Checks{
+				checker.PerResource(func(tc *checker.TestContext, pdb *policyv1beta1.PodDisruptionBudget) {
+					ingressGatewaysEnabled := checker.MustRenderValue[bool](tc, ".Values.ingressGateways.enabled")
+
+					if !ingressGatewaysEnabled {
+						return
+					}
+
+					ingressGatewaysPodDisruptionBudget, _ := checker.RenderValue[string](tc, ".Values.ingressGateways.podDisruptionBudget")
+					assert.Equal(tc.T, ingressGatewaysPodDisruptionBudget, pdb.Spec)
+				}),
+			},
+		},
+
+		{
+			Name: "Check .Values.kiali.enabled",
+			Covers: []string{
+				".Values.kiali.enabled",
+			},
+
+			Checks: test.Checks{
+				checker.PerWorkload(func(tc *checker.TestContext, obj metav1.Object, podTemplateSpec corev1.PodTemplateSpec) {
+					kialiEnabled := checker.MustRenderValue[bool](tc, ".Values.kiali.enabled")
+
+					if !kialiEnabled {
+						return
```

Comment:
There are a few tests like this which do essentially nothing. Every test should have at least one assert statement or theres no point to running the check at all since this could never fail.

Reply:
Kiali is actually a subchart. And this PR is only for Istio checks and not subcharts as there are issues while adding subcharts. That's why I kept it like this.